text/0000-orderer-snapshot.md

```diff
@@ -0,0 +1,263 @@
+6. prevBlockHash - technically not required, but this will make internal structure BlockchainInfo complete.
+
+Version 1 of the OSN snapshot described in this RFC defines two files as the content of the snapshot directory:
+1. Last config block for the channel. Name: "lastConfig.block"
+2. Metadata file containing all the rest of the snapshot content. Name: "metadata.json"
+
+Unlike peer snapshot the OSN snapshot does not contain state DB, private data configs, the list of TxIDs and commit hash - as the OSN is purposefully designed to be unaware of these. So these two kinds of snapshots (peer snapshot and OSN snapshot) bear resemblance only on the conceptual level, being completely different in the content.
+
+## Implementation considerations
+
+When a channel is joined from a snapshot by "channel joinbysnapshot" command, we have to save the following info on the block storage:
+1. Last config block from snapshot
+2. BootstrappingSnapshotInfo
+
+For BootstrappingSnapshotInfo persistence the implementation is already there, but for the last config block a new persistent unit of storage (a file in case of file-based ledger) should be created. Additionally all the procedures looking up the last config block should be modified to return the last config block from the snapshot in case the last config block is not found in the block ledger itself.
+
+The channel bootstrapping from snapshot must be crash-consistent, that is if a crash takes place while bootstrapping a channel, upon restart the orderer, either the bootstrapped channel exists in the usable state or not at all. In particular this means that the creation of the above two pieces of information (last config block and BootstrappingSnapshotInfo) must be an atomic transaction, and there should be no possibility after a crash to have one of them but not another.
+Possible implementations of such an atomic transaction may be keeping both pieces of information in the same file and creating the file atomically, or keeping them in separate files in the same directory and creating the directory atomically.
+
+## System channel co-existence
+
+The following API can work both with system channel and without it:
+- "channel list" - inherited from **channel participation** API
+
+The following APIs can work only without system channel, will fail in presence of system channel:
+- "channel joinbysnapshot"
+- "channel remove" - this one is not described in this RFC, it is inherited from **channel participation** API
+- "channel join" - inherited from **channel participation** API, deprecated
+- "snapshot submitrequest"
+
+## New channel creation ("genesis" snapshot)
+
+To be able to create a new channel we need a valid snapshot created with the genesis block as a last config block and lastBlockNum set to 0. We propose to modify *configtxgen* tool so that it can produce this "genesis" snapshot. Then we can add new orderers to the channel from the genesis block using this "genesis" snapshot.
```

Comment:
I return to the question of why we are making new APIs around joining by snapshot.

If we simply accept 'zero' values as defaults, then joining by a genesis block would specify the genesis block, would implicitly set the prune height to 0 (the genesis block), and implicitly set the previous block hash to nil, as there is no previous block.  Everything seems to work without changing output artifact formats.

Reply:
As we reverted the deprecation of "channel join" API, there is no need anymore in an additional tool support for the "genesis" snapshot. So now the RFC allows for the special case of "genesis" snapshot, but does not mandate any tool support for its creation.
Here:
0737f544deb4383c1d07a8f7368942869b3ba82a